### PR TITLE
Unify all the tox jobs into the top-level

### PR DESCRIPTION
**Pull request by addyess**
Unifies to a single pyproject.toml and applies linter rules to all `.py` files

Depends on #51 to be first merged

### CODE REVIEW

**Comment by addyess** [comment]
@louiseschmidtgen I got around to removing all those extraneous `pyproject.toml` files

File: .github/workflows/tests-k8s.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-name: Tests K8s
+name: Charm Analysis
```
Comment (louiseschmidtgen) [inline .github/workflows/tests-k8s.yaml:1]: Should we change the name of the file to `charm-analysis.yaml` and keep things consistent?

**Review by louiseschmidtgen — APPROVED** [review]
Thanks for cleaning up all these files,looks like we cover the COS code as well now! Nice!
Overall, LGTM I only have a rename suggestion.

File: .github/workflows/tests-k8s.yaml (hunk 1 above)
Comment (addyess) [reply .github/workflows/tests-k8s.yaml:1]: as you wish

**Pull request merged**